swift.en.md

```diff
@@ -0,0 +1,340 @@
+- **[MUST]** Add linebreak to end of files
+- **[MUST]** Place open brackets on same lines
+
+```swift
+// Bad
+func update() 
+{
+}
+
+// Good
+func update() {
+}
+```
+
+- **[MUST]** Use trailing comma on elements of collections
+    - Because of reducing file diffs
+
+```swift
+// Bad
+let ingredients = [
+    "bacon": "100g",
+    "lettuce": "1/4",
+    "tomato": "half"
+]
+
+// Good
+let ingredients = [
+    "bacon": "100g",
+    "lettuce": "1/4",
+    "tomato": "half",
+]
+```
+
+- **[MUST]** Do not use `()` wraps condition statements
+
+```swift
+// Bad
+if (a == 0) {
+}
+
+// Good
+if a == 0 {
+}
+```
+
+- **[MUST]** Do not use `;` on end of lines
+
+## Class, Struct
+
+- **[MUST]** Do not inherit `NSObject` except if needed to refer from Objective-C.
+- **[SHOULD]** Do not use `mutating` attribute for structs.
+    - Because structs should be immutable.
+
+## Method
+
+- **[MUST]** Use closest access level as possible.
+- **[MUST]** Do not use `Void` as returning type.
+
+```swift
+// Bad
+func update() -> Void {
+}
+
+// Good
+func update() {
+}
+```
+
+## Closure
+
+- **[MUST]** Use trailing closure if the closure has just one argument.
+
+```swift
+// Bad
+client.search(keyword: "Sushi", completion: { recipes in
+})
+
+// Good
+client.search(keyword: "Sushi") { recipes in
+}
+```
+
+- **[MUST]** Do not use `()` when calling the closure has just one argument.
+
+```swift
+// Bad
+let titles = recipes.map() { $0.title }
+
+// Good
+let titles = recipes.map { $0.title }
+```
+
+- **[MUST]** Use shorthand argument if closure defined in one line.
+    - In contrast, Do not use it if closure defined in multiple lines.
+
+```swift
+// Bad
+let titles = recipes.map { recipe in recipe.title }
+
+// Good
+let titles = recipes.map { $0.title }
+```
+
+```swift
+// Bad
+client.search(keyword: "Sushi") {
+    if $1 != nil {
+        recipes = $0
+    }
+}
+
+// Good
+client.search(keyword: "Sushi") { results, error in
+    if error != nil {
+        recipes = results
+    }
+}
+```
+
+- **[MUST]** Do not use `@escaping` without needed
+- **[MUST]** Use `weak` instead of `unowned` as capture controls.
+    - In some cases, `unowned` improves performance. In other hand readability will be lost.
+
+## Property, Variable
+
+- **[MUST]** Use `let` to make variables immutable as possible.
+- **[MUST]** Drop type definition as possible. However, in the case of avoiding type inference for improvement compilation time, you can define clearly.
+
+### Property
+
+- **[MUST]** Use closest access control as possible.
+- **[MUST]** Use `private(set)` for the property want to write from limited scopes.
+- **[MUST]** Use `weak` as property attributes as possible.
+    - You should use `weak` to avoid circular references (eg. `@IBOutlet`, delegate)
+- **[MUST]** Do not use `unowned` as property attribute.
+    - In some cases, `unowned` improves performance. In other hand readability will be lost.
```

Comment:
And also in the sense that we know exactly that this reference is gonna outlive you. Use `unowned` is a perfect keyword for that.